core/src/nodes/collection.rs

```diff
@@ -58,6 +54,9 @@ where
     }
 }
 
+impl<TInEvent, TOutEvent, THandler, TReachErr, THandlerErr, TUserData, TConnInfo, TPeerId> Unpin for
+    CollectionStream<TInEvent, TOutEvent, THandler, TReachErr, THandlerErr, TUserData, TConnInfo, TPeerId> { }
```

Comment:
Otherwise `Unpin` is implemented automatically only if all the fields are `Unpin` as well.
It is never a bad idea to implement `Unpin` for a type. The reason why we do it for this type is because it implements the `Stream` trait.